compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2249,21 +2252,76 @@ public void visit(BLangRecordLiteral recordLiteral) {
                 BLangExpression spreadOpExpr = spreadOpField.expr;
 
                 analyzeExpr(spreadOpExpr);
-                if (spreadOpExpr.type.tag != TypeTags.RECORD) {
+
+                BType spreadOpExprType = spreadOpExpr.type;
+                int spreadFieldTypeTag = spreadOpExprType.tag;
+                if (spreadFieldTypeTag == TypeTags.MAP) {
+                    if (inclusiveTypeSpreadField != null) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MULTIPLE_INCLUSIVE_TYPES);
+                        break;
+                    } else {
+                        inclusiveTypeSpreadField = spreadOpField;
+                    }
+                    if (fields.size() > 1) {
+                        if (names.size() > 0) {
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.INCLUSIVE_MAPPING_TYPE_IN_SPREAD_FIELD,
+                                spreadOpExpr);
+                            break;
+                        }
+                        // Skipping to avoid multiple error messages
+                        continue;
+                    }
+                }
+
+                if (spreadFieldTypeTag != TypeTags.RECORD) {
                     continue;
                 }
 
-                for (BField bField : ((BRecordType) spreadOpExpr.type).fields.values()) {
-                    if (Symbols.isOptional(bField.symbol)) {
-                        continue;
+                BRecordType spreadExprRecordType = (BRecordType) spreadOpExprType;
+                boolean isSpreadExprRecordTypeSealed = spreadExprRecordType.sealed;
+                if (!isSpreadExprRecordTypeSealed) {
+                    // More than one inclusive-type-descriptors with spread-fields are not allowed.
+                    if (inclusiveTypeSpreadField != null) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MULTIPLE_INCLUSIVE_TYPES);
+                        break;
```

Comment:
Shouldn't we let the rest of the checks run here, without breaking? 

For example, in the following scenario we can give the 
`invalid usage of record literal: duplicate key 'i' via spread operator '...r2'` error too, right?

```ballerina
public function main() {
    record {
        int i;
    } r1 = {i: 1};

    record {
        int i;
    } r2 = {i: 1};  

    var x = {...r1, ...r2};
}
```